lib/list.gi

```diff
@@ -3371,7 +3371,11 @@ InstallOtherMethod( \*,
           if depthDiff < -1 or depthDiff > 1 then
               return PROD_LISTS_SPECIAL(left, right, depth1, depth2 );
           else
-              return PROD_LIST_LIST_DEFAULT( left, right, depthDiff );
+              if IsEmpty( left ) then
+                  return [];
```

Comment:
I kind of expected it to fail, but missed to check that. GAP throws an error when the right list is empty.
```
gap> l := [1,2,3];
[ 1, 2, 3 ]
gap> l * [];
Error, Inner product multiplication of lists: no summands in
  return PROD_LIST_LIST_DEFAULT( left, right, depthDiff )
 ; at /opt/gap/gap4r8/lib/list.gi:3250 called from 
<function "unknown">( <arguments> )
 called from read-eval loop at line 2 of *stdin*
```